File: io/lucamienert/basic/ast/Line.java
package io.lucamienert.basic.ast;

import io.lucamienert.basic.instruction.Instruction;
import io.lucamienert.basic.instruction.InstructionSequence;
import io.lucamienert.basic.instruction.OpCode;

public class Line {

    private final int number;
    private final Statement statement;

    public Line(int number, Statement statement) {
        this.number = number;
        this.statement = statement;
    }

    public int getNumber() {
        return number;
    }

    public Statement getStatement() {
        return statement;
    }

    public void compile(InstructionSequence seq) {
        seq.append(new Instruction(OpCode.LABEL, seq.createLineLabel(number)));
        statement.compile(seq);
    }
    
}

File: BasicCompiler/Ast/Line.cs
using BasicCompiler.Instructions;

namespace BasicCompiler.Ast;

public class Line
{
    public int Number { get; set; }
    public Statement Statement { get; set; }

    public Line(int number, Statement statement)
    {
        Number = number;
        Statement = statement;
    }

    public void Compile(InstructionSequence instructionSequence)
    {
        instructionSequence.Append(new Instruction(OpCode.LABEL, InstructionSequence.CreateLineLabel(Number)));
        Statement.Compile(instructionSequence);
    }
}
